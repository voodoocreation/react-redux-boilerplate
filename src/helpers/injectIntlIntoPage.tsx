import { NextComponentType } from "next";
import * as React from "react";
import { injectIntl } from "react-intl";

import { IPageContext } from "../components/connected/App/App";

export default <T extends NextComponentType<IPageContext, any, any>>(
  Component: T
) => {
  const IntlPage = injectIntl(Component);

  class ComponentWithIntl extends React.Component<{}> {
    public static getInitialProps = async (context: IPageContext) => {
      let props;

      if (typeof Component.getInitialProps === "function") {
        props = await Component.getInitialProps(context);
      }

      return { ...props };
    };

    public render() {
      return <IntlPage {...this.props} />;
    }
  }

  return ComponentWithIntl;
};
