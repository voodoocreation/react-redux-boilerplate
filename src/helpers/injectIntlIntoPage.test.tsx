import { mount } from "enzyme";
import * as React from "react";

import injectIntlIntoPage from "./injectIntlIntoPage";

describe("[helpers] injectIntl", () => {
  it("injects intl into next.js page component correctly when `getInitialProps` is defined", async () => {
    class WithInitialProps extends React.Component<{ test: string }> {
      public static defaultProps = { test: "From defaultProps" };

      public static async getInitialProps() {
        return { test: "From getInitialProps()" };
      }

      public render() {
        return <div>{this.props.test}</div>;
      }
    }

    const WithInitialPropsWrapped = injectIntlIntoPage(WithInitialProps);
    const initialProps = await WithInitialPropsWrapped.getInitialProps({});
    const actual = mount(<WithInitialPropsWrapped {...initialProps} />);

    expect(initialProps).toEqual({ test: "From getInitialProps()" });
    expect(actual.render()).toMatchSnapshot();
  });

  it("injects intl into next.js page component correctly when `getInitialProps` isn't defined", async () => {
    // tslint:disable-next-line
    class WithoutInitialProps extends React.Component<{ test: string }> {
      public static defaultProps = { test: "From defaultProps" };

      public render() {
        return <div>{this.props.test}</div>;
      }
    }

    const WithoutInitialPropsWrapped = injectIntlIntoPage(WithoutInitialProps);
    const initialProps = await WithoutInitialPropsWrapped.getInitialProps({});
    const actual = mount(<WithoutInitialPropsWrapped {...initialProps} />);

    expect(initialProps).toEqual({});
    expect(actual.render()).toMatchSnapshot();
  });
});
