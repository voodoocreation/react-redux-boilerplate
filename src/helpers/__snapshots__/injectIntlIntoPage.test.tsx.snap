// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[helpers] injectIntl injects intl into next.js page component correctly when \`getInitialProps\` is defined 1`] = `
<div>
  From getInitialProps()
</div>
`;

exports[`[helpers] injectIntl injects intl into next.js page component correctly when \`getInitialProps\` isn't defined 1`] = `
<div>
  From defaultProps
</div>
`;
